Pod/Classes/AirshipAdapter.swift

```diff
@@ -171,5 +222,47 @@ private class AirshipGimbalDelegate : NSObject, PlaceManagerDelegate {
     func placeManager(_ manager: PlaceManager, didDetect location: CLLocation) {
         AirshipGimbalAdapter.shared.delegate?.placeManager?(manager, didDetect: location)
     }
+    
+    private func trackPlaceEventFor(_ visit: Visit, boundaryEvent: UABoundaryEvent) {
+        if shouldCreateRegionEvents,
+           let regionEvent = RegionEvent(regionID: visit.place.identifier,
+                                           source: source,
+                                    boundaryEvent: boundaryEvent) {
+            Airship.analytics.addEvent(regionEvent)
+        }
+
+        if boundaryEvent == .enter, shouldCreateCustomEntryEvent {
+            createAndTrackEvent(withName: customEntryEventName, forVisit: visit, boundaryEvent: boundaryEvent)
+        } else if boundaryEvent == .exit, shouldCreateCustomExitEvent {
+            createAndTrackEvent(withName: customExitEventName, forVisit: visit, boundaryEvent: boundaryEvent)
+        }
+    }
+    
+    private func createAndTrackEvent(withName eventName: String,
+                                     forVisit visit: Visit,
+                                     boundaryEvent: UABoundaryEvent) {
+        // create event properties
+        var visitProperties:[String : Any] = [:]
+        visitProperties["visitID"] = visit.visitID
+        visitProperties["placeIdentifier"] = visit.place.identifier
+        visitProperties["placeName"] = visit.place.name
+        visitProperties["source"] = source
+        visitProperties["boundaryEvent"] = boundaryEvent.rawValue
+        var placeAttributes = Dictionary<String, Any>()
+        for attributeKey in visit.place.attributes.allKeys() {
+            if let value = visit.place.attributes.string(forKey: attributeKey) {
+                placeAttributes[attributeKey] = value
+                visitProperties.updateValue(value, forKey: "GMBL_PA_\(attributeKey)")
+            }
+        }
+        visitProperties["placeAttributes"] = placeAttributes
```

Comment:
Leaving a comment here just because this is not the solution we did in android. Can't remember if we decided to have different solutions for both though.